database/query/repositories.sql

```diff
@@ -45,6 +49,28 @@ WHERE project_id = $1 AND webhook_id IS NOT NULL
     AND (lower(provider) = lower(sqlc.narg('provider')::text) OR sqlc.narg('provider')::text IS NULL)
 ORDER BY repo_name;
 
+-- name: ListEligibleRepositoriesAfterID :many
+SELECT r.* FROM repositories r
+  INNER JOIN rule_evaluations re ON re.repository_id = r.id
+  INNER JOIN rule_details_eval rde ON rde.rule_eval_id = re.id
+WHERE r.id > $1
+GROUP BY r.id
+HAVING MIN(rde.last_updated) + sqlc.arg('min_elapsed')::interval < NOW()
+ORDER BY r.id
+LIMIT sqlc.narg('limit')::bigint;
```

Comment:
Agreed on the statement analysis. The current query (without the sub-select) guarantees that we get `limit` rows (if they are there). With a sub-select, selected rows might not be eligible i.e. `last_updated` was recent. 

Agreed with the point that this will be a single time-consuming query rather than a steady load form of query. To keep things and mocking simpler on the application side, I'd go with a sub-query rather than two different queries, i.e. selection and filtration.  

Just using `reminder_last_sent` isn't a good parameter IMO. We can have a case where a reminder was sent 24h ago but the repo was recently updated (edge based), so this would result in extra load on the server (reconciling the repo).